scripts/dev_setup.sh

```diff
@@ -428,6 +436,28 @@ function install_cvc4 {
   rm -rf "$TMPFILE"
 }
 
+function install_codegen_deps {
+    if [[ "$PACKAGE_MANAGER" != "apt-get" ]]; then
+      # Not supported.
+      return
+    fi
+    echo "deb http://http.us.debian.org/debian/ buster-backports main" | "${PRE_COMMAND[@]}" tee -a /etc/apt/sources.list
+    echo "deb-src http://http.us.debian.org/debian/ buster-backports main" | "${PRE_COMMAND[@]}" tee -a /etc/apt/sources.list
+    "${PRE_COMMAND[@]}" apt-get update
+    "${PRE_COMMAND[@]}" apt-get upgrade -y
+    "${PRE_COMMAND[@]}" apt-get install -y apt-transport-https python3-all-dev python3-pip clang llvm default-jdk golang-1.14-go/buster-backports
+    "${PRE_COMMAND[@]}" ln -sf /usr/lib/go-1.14 /usr/lib/golang
+    "${PRE_COMMAND[@]}" python3 -m pip install pyre-check==0.0.59
+    "${PRE_COMMAND[@]}" python3 -m pip install numpy==1.20.1
+    "${PRE_COMMAND[@]}" curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.3/install.sh | bash
```

Comment:
Again since we do reruns of this script, check to see if the tool is available at the right version before install

Reply:
I believe these operations are idempotent, except for NVM -- which I can try to replace by debian package.

Reply:
I now use proper debian packages for node and npm instead of manually installing nvm.